elephant/spectral.py

```diff
@@ -523,6 +529,543 @@ def multitaper_psd(signal, n_segments=1, len_segment=None,
+        n_per_seg = int(data.shape[axis] /
+                        (n_segments - overlap * (n_segments - 1)))
+
+    n_overlap = int(n_per_seg * overlap)
+    n_overlap_step = n_per_seg - n_overlap
+    n_segments = int((length_signal - n_overlap) / (n_per_seg - n_overlap))
+
+    # Generate frequencies for spectral measure estimate
+    if func_params_dict.get('return_onesided'):
+        freqs = np.fft.rfftfreq(n_per_seg, d=1/fs)
+    else:
+        freqs = np.fft.fftfreq(n_per_seg, d=1/fs)
+
+    # Zero-pad signal to fit segment length
+    remainder = length_signal % n_overlap_step
+
+    data = np.pad(data, [(0, 0), (0, remainder)],
+                  mode='constant', constant_values=0)
+
+    # Generate array for storing cross spectra estimates of segments
+    seg_estimates = np.zeros((n_segments,
+                              data.shape[0],
+                              data.shape[0],
+                              len(freqs)),
+                             dtype=np.complex64)
+
+    n_overlap_step = n_per_seg - n_overlap
+
+    for i in range(n_segments):
+
+        _, estimate = func(
+            data[:, i * n_overlap_step:i * n_overlap_step + n_per_seg],
+            **func_params_dict)
+
+        seg_estimates[i] = estimate
+
+    avg_estimate = np.mean(seg_estimates, axis=0)
+
+    return freqs, avg_estimate
+
+
+def segmented_multitaper_cross_spectrum(signals, n_segments=1,
+                                        len_segment=None,
+                                        frequency_resolution=None, overlap=0.5,
+                                        fs=1, nw=4, num_tapers=None,
+                                        peak_resolution=None,
+                                        return_onesided=True):
+    """
+    Estimates the cross spectrum of a given 'neo.AnalogSignal' using the
+    Multitaper method on segments of the data.
+
+    The cross spectrum is obtained through the following steps
+
+    1. Cut the given data into several overlapping segments. The degree of
+       overlap can be specified by parameter `overlap` (default is 0.5,
+       i.e. segments are overlapped by the half of their length).
+       The number and the length of the segments are determined according
+       to the parameters `n_segments`, `len_segment` or `frequency_resolution`.
+       By default, the data is not cut into  segments;
+
+    2. Calculate 'num_tapers' approximately independent estimates of the
+       spectrum by multiplying the signal with the discrete prolate spheroidal
+       functions (also known as Slepian function) and calculate the PSD of each
+       tapered segment
+
+    3. Average the approximately independent estimates of each segment to
+       decrease overall variance of the estimates
+
+    4. Average the obtained estimates for each segment
+
+    The estimation is done by applying `_segmented_apply_func` to
+    `multitaper_cross_spectrum`. See the respective functions for further
```

Comment:
Whenever there is a reference to another function, reference with :func: followed by the name in backticks to allow links in the documentation page (e.g., :func:\`multitaper_cross_spectrum\`). The function reference should also be added in a `See Also` section.

Reply:
Addressed in https://github.com/NeuralEnsemble/elephant/pull/525/commits/5ddd042c88783fdc4203819a36146c746510599d